packages/gatsby/src/commands/__tests__/build-utils.ts

```diff
@@ -4,14 +4,21 @@ import {
   IHtmlFileState,
   IStaticQueryResultState,
 } from "../../redux/types"
-import { calcDirtyHtmlFiles } from "../build-utils"
 
+const platformSpy = jest.spyOn(require(`os`), `platform`)
 interface IMinimalStateSliceForTest {
   html: IGatsbyState["html"]
   pages: IGatsbyState["pages"]
 }
 
 describe(`calcDirtyHtmlFiles`, () => {
+  let calcDirtyHtmlFiles
+  beforeEach(() => {
+    jest.isolateModules(() => {
+      calcDirtyHtmlFiles = require(`../build-utils`).calcDirtyHtmlFiles
```

Comment:
Why do we need to isolate the module here (and also in the test)? I never used `isolateModules` before so just curious.

Reply:
I do need to isolate because of `os.platform()` calls (that I _need_ to mock) execute at import time (that's why I do have them isolated in casing specific tests). 

In here (swiching import that is already there to this) it just seemed that if I import regularly before I import in `isolateModules` just for casing tests - I still get cached module which means my mocking of `os.platform()` has no effect